### PR TITLE
Some player book refactors

### PR DESCRIPTION
**Pull request by bsxf-47**
Book pages now have their own classes, and all interaction with the player book is done through a global PlayerBook class instance.

I'd like to do more here but the patchset is rather big as it is.

Oh and if you have any better ideas for some of the commits' titles, please let me know. Still sometimes unsure about those.

### CODE REVIEW

File: src/gui/book/Book.cpp [hunk 1]
```diff
@@ -1449,13 +1457,6 @@ void ARX_INTERFACE_ManageOpenedBook() {
 	} else {
 		g_playerBook.stats.renderNewQuest();
 	}
-	
-	GRenderer->GetTextureStage(0)->setMinFilter(TextureStage::FilterNearest);
-	GRenderer->GetTextureStage(0)->setMagFilter(TextureStage::FilterNearest);
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: I'm not so sure about removing this.

File: src/gui/book/Book.cpp [hunk 2]
```diff
@@ -1448,7 +1454,7 @@ void ARX_INTERFACE_ManageOpenedBook() {
 				break;
 			}
 			case BOOKMODE_QUESTS: {
-				gui::manageQuestBook();
+				g_playerBook.journal.render();
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: The game text (tutorial, key binding, etc.) refers to this as the quest book so maybe we should keep the name. Not entirely sure though.

File: src/gui/book/Book.cpp [hunk 3]
```diff
@@ -72,8 +72,12 @@ class SpellsPage : public PlayerBookPage {
 
 class MapPage : public PlayerBookPage {
 public:
+	MapPage();
 	void render();
+	void setMapLevel(long level);
 private:
+	long currentLevel;
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: Missing m_ prefix.

File: src/gui/book/Book.cpp [hunk 4]
```diff
@@ -64,8 +64,11 @@ class StatsPage : public PlayerBookPage {
 
 class SpellsPage : public PlayerBookPage {
 public:
+	SpellsPage();
 	void render();
 private:
+	long currentTab;
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: Missing m_ prefix.

File: src/gui/book/Book.cpp [hunk 5]
```diff
@@ -92,6 +92,8 @@ class JournalPage : public PlayerBookPage {
 public:
 	void render();
 private:
+	Note questBook;
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: m_ prefix?

File: src/gui/book/Book.cpp [hunk 6]
```diff
@@ -1696,5 +1672,21 @@ void MapPage::ARX_INTERFACE_ManageOpenedBook_LeftTabs_Map() {
...
+			if(!quest.empty()) {
+				text += quest;
+				text += "\n\n";
+			}
+		}
+		questBook.setData(Note::QuestBook, text);
+		questBook.setPage(questBook.pageCount() - 1);
+	}
+
+	manageNoteActions(questBook);
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: This is all not really about rendering but about managing the state of the quest book.

File: src/gui/book/Book.cpp [hunk 7]
```diff
@@ -51,7 +51,7 @@ class PlayerBookPage {
 public:
 	void playReleaseSound();
 	void playErrorSound();
-	void ARX_INTERFACE_ManageOpenedBook_LeftTabs(bool tabVisibility[10], long & activeTab);
+	void drawLeftTabsCommon(bool tabVisibility[10], long & activeTab);
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: This is also updates state besides drawing so the function name is a bit misleading - maybe it should be split?

File: src/gui/book/Book.cpp [hunk 8]
```diff
@@ -64,7 +64,7 @@ class StatsPage : public PlayerBookPage {
 	void RenderBookPlayerCharacter();
 	bool CheckAttributeClick(Vec2f pos, float * val, TextureContainer * tc);
 	bool CheckSkillClick(Vec2f pos, float * val, TextureContainer * tc, float * oldval);
-	Color attrubuteModToColor(float modValue, float baseValue = 0.f);
+	Color attributeModToColor(float modValue, float baseValue = 0.f);
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: Forgot to merge this with the earlier commit?

File: src/gui/book/Book.cpp [hunk 9]
```diff
@@ -382,7 +377,7 @@ void PlayerBook::openPage(ARX_INTERFACE_BOOK_MODE newPage, bool toggle) {
 
 	} else {
 		// Otherwise open the book
-		ARX_INTERFACE_BookToggle();
+		this->toggle();
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: Unnecessary `this->`.

File: tools/savetool/SaveView.cpp [hunk 10]
```diff
@@ -858,7 +858,7 @@ static int view_player(SaveBlock & save, const char * dat, size_t size) {
 	if(asp.Interface) {
 		std::cout << "Interface:";
 		s16 interface_flags = asp.Interface;
-		print_flag(interface_flags, INTER_MAP, "map");
+		print_flag(interface_flags, INTER_PLAYERBOOK, "map");
```
Comment (dscharrer) [inline tools/savetool/SaveView.cpp]: You can also change the string here since it's just debug information.

File: src/gui/book/Book.cpp [hunk 11]
```diff
@@ -117,107 +117,102 @@ void PlayerBook::clearJournal() {
...
+		}
+		case 1: {
+			Vec2f pos = Vec2f(98.f, 112.f);
+			Vec2f activePos = Vec2f(100.f, 114.f);
+
+			drawLeftTabOneCommon(tabNum, activeTab, pos, activePos);
+			break;
+		}
+		case 2:
+		{
```
Comment (dscharrer) [inline src/gui/book/Book.cpp]: Braces should be on the same line as the case statement.

File: src/gui/book/Book.cpp (hunk 9 above)
Comment (bsxf-47) [reply src/gui/book/Book.cpp]: Not really, the member function's parameter is also named `toggle`. Shall I rename it?

File: src/gui/book/Book.cpp [hunk 12]
```diff
@@ -47,6 +47,23 @@
...
+class StatsPage {
+public:
+	void render();
+	void renderNewQuest();
+private:
+	void ARX_INTERFACE_ManageOpenedBook_Stats();
+	void RenderBookPlayerCharacter();
+	bool CheckAttributeClick(Vec2f pos, float * val, TextureContainer * tc);
+	bool CheckSkillClick(Vec2f pos, float * val, TextureContainer * tc, float * oldval);
+	Color attrubuteModToColor(float modValue, float baseValue = 0.f);
```
[the remark this answers is not in the document]
Comment (bsxf-47) [reply src/gui/book/Book.cpp]: This is actually the original spelling, and the later commit is a fix.

File: src/gui/book/Book.cpp (hunk 8 above)
Comment (bsxf-47) [reply src/gui/book/Book.cpp]: This is just a fix of the original spelling, I thought doing the rename in addition to what the commit does was a bit too much for one commit.

File: src/gui/book/Book.cpp (hunk 1 above)
Comment (bsxf-47) [reply src/gui/book/Book.cpp]: You're right, not sure what I was thinking there. Readded.

File: src/gui/book/Book.cpp (hunk 9 above)
Comment (dscharrer) [reply src/gui/book/Book.cpp]: Yeah, renaming would be good.

File: src/gui/book/Book.cpp (hunk 12 above)
[the remark this answers is not in the document]
Comment (dscharrer) [reply src/gui/book/Book.cpp]: Hm, ok.

**Comment by bsxf-47** [comment]
All the requested changes should now be in.

**Pull request merged**